src/content/en/tools/workbox/guides/advanced-recipes.md

```diff
@@ -0,0 +1,173 @@
+project_path: /web/tools/workbox/_project.yaml
+book_path: /web/tools/workbox/_book.yaml
+description: Advanced recipes to use with Workbox.
+
+{# wf_updated_on: 2017-12-17 #}
+{# wf_published_on: 2017-12-17 #}
+{# wf_blink_components: N/A #}
+
+# Advanced Recipes {: .page-title }
+
+{% include "web/tools/workbox/_shared/alpha.html" %}
+
+## Offer a page reload for users
+
+A common UX pattern for progressive web apps is to show a banner when a service
+worker has updated and waiting to install.
+
+To do this you'll need to add some code to your page and to your service worker.
+
+**Add to your page**
+
+```javascript
+const showRefreshUI = (registration) => {
```

Comment:
I tend to avoid arrow functions and `const` in code that's meant to run inside of a `window` context, as it breaks parsing the JS in older browsers, and not everyone transpiles.